test/realtimecontroller/DefaultRealtimeController.test.ts

```diff
@@ -538,6 +544,44 @@ describe('DefaultRealtimeController', () => {
       expect(callbackFired).to.be.true;
     });
 
+    it('will unsubscribe from a given callback if provided', () => {
+      const rt: RealtimeController = new DefaultRealtimeController();
+      const fooAttendeeId = 'foo-attendee';
+      const sentVolume = 0.5;
+      const sentMuted = false;
+      const sentSignalStrength = 1;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const state: RealtimeState = ((rt as unknown) as any).state as RealtimeState;
+      state.volumeIndicatorCallbacks = {};
+      const callback1 = (
+        attendeeId: string,
+        volume: number | null,
+        muted: boolean | null,
+        signalStrength: number | null
+      ): void => {
+        expect(attendeeId).to.equal(fooAttendeeId);
+        expect(volume).to.equal(sentVolume);
+        expect(muted).to.equal(sentMuted);
+        expect(signalStrength).to.equal(sentSignalStrength);
+      };
+      const callback2 = (
+        attendeeId: string,
+        volume: number | null,
+        muted: boolean | null,
+        signalStrength: number | null
+      ): void => {
+        expect(attendeeId).to.equal(fooAttendeeId);
+        expect(volume).to.not.equal('');
+        expect(muted).to.not.equal('');
+        expect(signalStrength).to.not.equal('');
+      };
+      rt.realtimeSubscribeToVolumeIndicator(fooAttendeeId, callback1);
```

Comment:
What happens when the `callback1` and `callback2` are anonymous functions? We should check if deletion works fine for anonymous functions as well.

Reply:
This pattern is already used in some other places as well. And as we looked yesterday, I wouldn't expect that to work.

Reply:
Strictly speaking, `callback1` and `callback2` _are_ anonymous functions — all arrow functions are anonymous, as opposed to `function foo() {}`, which is named.

I think what you mean is "functions that are not referenced anywhere else", e.g.,

```typescript
rt.realtimeSubscribeToVolumeIndicator('foo', () => {
  // whatever
});
```

In that case there's nothing you can do — you have no reference to the callback.

Reply:
That makes sense. Thank you for the clarity.